src/writer/cluster.h

```diff
@@ -88,7 +88,7 @@ class Cluster {
     ClusterData _data;
     mutable Blob compressed_data;
     std::string tmp_filename;
-    mutable pthread_mutex_t m_closedMutex;
+    mutable std::mutex m_closedMutex;
     bool closed = false;
```

Comment:
Isn't it better to use `std::atomic<bool>` instead (and get rid of the mutex)?

Reply:
Yes, this is done.